pkg/daemon/ceph/osd/volume.go

```diff
@@ -396,6 +396,11 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 	metadataDevices := make(map[string]map[string]string)
 	for name, device := range devices.Entries {
 		if device.Data == -1 {
+			if device.Metadata != nil {
+				logger.Infof("skipping metadata device %s config since it will be configured with a data device", name)
```

Comment:
Is this the right behavior? I would think we should do nothing with the device, as this is clearly a configuration that isn't properly specified. Maybe we should even stop OSD provisioning entirely. ?

Reply:
The `for` loop is expected to iterate over all the data devices. But metadata devices were also added to the `devices` map, which was causing the metadataDevice to also be treated as a data device. It's really a bug in the code rather than a configuration problem. All this code that translates the settings into c-v parameters is very confusing. I'd like to do a more thorough review of it later, but for now this was fixing the c-v failure blocking any metadataDevice OSDs from being created.

Reply:
Makes sense... but the logic seems backwards IIUC. Should the device be skipped instead? The metadata devices are specified specifically right, but the devices are added automatically, so if a device is in both, the metadata device should be chosen but not the data one to my logic.

Reply:
This loop is only intended to loop over data devices. If there is a metadataDevice specified, it will be looked up and handled below starting on line 418. Really, the metadataDevice should not have been added to the `devices` map in the first place, but I'd like to review that change with a more in-depth look at this code to see if it can be simplified.

Reply:
Okay! My bad! I was misinterpreting! LGTM, and thanks!